### PR TITLE
Let's review our git workflow here. This might be a long running PR

### PR DESCRIPTION
**Pull request by dreinavarro**

### CODE REVIEW

File: general/git.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Git & GitHub
+For source control we use GitHub. All projects are repositories under [scrambledeggs](https://github.com/scrambledeggs).
+
+## Workflow
+- Prefer to adapt [Gitlab flow](https://about.gitlab.com/2014/09/29/gitlab-flow/)
+- The canonical version of the code for active development lives in `master`
+- When a new feature is worked on, a branch will be created from `master`
+- Use descriptive feature branch names and name them using [`kebab-case`](https://en.wikipedia.org/wiki/Letter_case#Special_case_styles)
+- For fixes, treat the same as features but add `fix` to branch name i.e `fix-your-face`
```
Comment (alecguintu) [inline general/git.md:9]: Should modifications (eg. copy changes) be `fix-` as well?

File: general/git.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  - For deployed apps, squash commit to `production` branch and tag it with version number when applicable
+
+## Versioning
+- Prefer to use [semantic versioning](http://semver.org/) where applicable.
+
+## Commits and Pull Requests
+- Commit often. Small commits are acceptable. General guideline is to commit when tests are passing.
+- Make your commits contextual. Avoid commits that does other things that the commit was intended to.
+- Commit messages that makes sense. Usually in this pattern: `"{action} {purpose|reason} {target}"`, order may vary.
+  Avoid:
```
Comment (alecguintu) [inline general/git.md:24]: Hahaha! I have a lot of this.

Comment (alecguintu) [reply general/git.md:24]: Though if it's under a branch already, should you still be mentioning it in the commit message?

branch: fix-homepage-downcase-title
Commit message: Fix Homepage's downcase title?

File: general/git.md (hunk 1 above)
Comment (dreinavarro) [reply general/git.md:9]: I guess we can make a specific label for copy-changes like `copy-homescreen` etc. What do you think?

File: general/git.md (hunk 2 above)
Comment (dreinavarro) [reply general/git.md:24]: Commits are more specific, branch names are more general.

Comment (dreinavarro) [reply general/git.md:24]: @alecguintu for small commits probably this would happen. What's more important is following the process and habit.

**Comment by dreinavarro** [comment]
@alecguintu @angelofrancisco need initial approval

File: general/git.md (hunk 1 above)
Comment (angelofrancisco) [reply general/git.md:9]: What about for branches like `release`? Or do we avoid making a branch for release preparations, database packaging etc?

Comment (angelofrancisco) [reply general/git.md:9]: Nevermind, disregard since `release` is discussed below

File: general/git.md (hunk 2 above)
Comment (angelofrancisco) [reply general/git.md:24]: Just to clarify, which branch is allowed to merge to master - `dev`,`release`, or `production`?

Comment (dreinavarro) [reply general/git.md:24]: @angelofrancisco features branches we are working on are merged to `master`. Once okay on tests, we branch from `master` to a `release` branch.

Comment (angelofrancisco) [reply general/git.md:24]: So no more dev branch @dreinavarro?

Comment (dreinavarro) [reply general/git.md:24]: `master` is the "`dev`" branch. does that make sense? can we find a good example for this that has a good workflow

Comment (dreinavarro) [reply general/git.md:24]: Basically, builds for testing are created from master.

**Review by dreinavarro — COMMENTED** [review]
Discussion for app builds

**Review by angelofrancisco — APPROVED** [review]

**Comment by dreinavarro** [comment]
@alecguintu @angelofrancisco please review last commit. lets merge this version for now if all is well with you guys

**Pull request merged**